### DISCUSSION TITLE
Native flash message support ?

### DISCUSSION THREAD

**Discussion by zogs** [sveltejs/kit#7817] [Ideas]
Are flash messages something that is ultimatly planed, or been discused ?
Or do we have to find a way to implement it by ourself ? ( i see there is already one library about it -> https://www.npmjs.com/package/sveltekit-flash-message )

thx

**Comment by teemingc** [comment]
This is currently possible through the form actions API.
https://kit.svelte.dev/docs/form-actions#anatomy-of-an-action

After sending a request to a form action, you can return data through the form property on the page.
```html
<script lang="ts">
  import type { PageData, ActionData } from './$types';

  export let data: PageData;

  export let form: ActionData;
</script>

{#if form?.success}
  <!-- this message is ephemeral; it exists because the page was rendered in
       response to a form submission. it will vanish if the user reloads -->
  <p>Successfully logged in! Welcome back, {data.user.name}</p>
{/if}
```

**Comment by zogs** [comment]
Yes but this wont work with a redirect() method, so it can't be use as a global flash message system....
